src/Internal/Wallet.purs

```diff
@@ -280,3 +314,97 @@ dummySign tx@(Transaction { witnessSet: tws@(TransactionWitnessSet ws) }) =
               "ed25519_sig1ynufn5umzl746ekpjtzt2rf58ep0wg6mxpgyezh8vx0e8jpgm3kuu3tgm453wlz4rq5yjtth0fnj0ltxctaue0dgc2hwmysr9jvhjzswt86uk"
           )
     )
+
+getNetworkId :: Wallet -> Aff NetworkId
+getNetworkId wallet = do
+  actionBasedOnWallet
+    (\w -> intToNetworkId <=< _.getNetworkId w)
+    (\kw -> pure (unwrap kw).networkId)
+    wallet
+  where
+  intToNetworkId :: Int -> Aff NetworkId
+  intToNetworkId = case _ of
+    0 -> pure TestnetId
+    1 -> pure MainnetId
+    _ -> liftEffect $ throw "Unknown network id"
+
+getUnusedAddresses :: Wallet -> Aff (Array Address)
+getUnusedAddresses wallet = fold <$> do
+  actionBasedOnWallet _.getUnusedAddresses
+    (\_ -> pure $ pure [])
+    wallet
+
+getChangeAddress :: Wallet -> Aff (Maybe Address)
+getChangeAddress wallet = do
+  actionBasedOnWallet _.getChangeAddress
+    (\kw -> pure $ pure (unwrap kw).address)
+    wallet
+
+getRewardAddresses :: Wallet -> Aff (Array Address)
+getRewardAddresses wallet = fold <$> do
+  actionBasedOnWallet _.getRewardAddresses
+    (\kw -> pure $ pure $ pure (unwrap kw).address)
+    wallet
+
+getWalletAddresses :: Wallet -> Aff (Array Address)
+getWalletAddresses wallet = fold <$> do
+  actionBasedOnWallet _.getWalletAddresses
+    (\kw -> pure $ pure $ Array.singleton (unwrap kw).address)
+    wallet
+
+signData :: Address -> RawBytes -> Wallet -> Aff (Maybe DataSignature)
+signData address payload wallet = do
+  actionBasedOnWallet
+    (\w conn -> w.signData conn address payload)
+    (\kw -> pure <$> (unwrap kw).signData payload)
+    wallet
+
+actionBasedOnWallet
+  :: forall (m :: Type -> Type) (a :: Type)
+   . MonadAff m
+  => (Cip30Wallet -> Cip30Connection -> Aff a)
+  -> (KeyWallet -> m a)
+  -> Wallet
+  -> m a
+actionBasedOnWallet walletAction keyWalletAction =
+  case _ of
+    Eternl wallet -> liftAff $ callCip30Wallet wallet walletAction
+    Nami wallet -> liftAff $ callCip30Wallet wallet walletAction
+    Gero wallet -> liftAff $ callCip30Wallet wallet walletAction
+    Flint wallet -> liftAff $ callCip30Wallet wallet walletAction
+    Lode wallet -> liftAff $ callCip30Wallet wallet walletAction
+    NuFi wallet -> liftAff $ callCip30Wallet wallet walletAction
+    KeyWallet kw -> keyWalletAction kw
+
+ownPubKeyHashes :: Wallet -> Aff (Array PubKeyHash)
+ownPubKeyHashes wallet = Array.catMaybes <$> do
+  getWalletAddresses wallet >>= traverse \address -> do
+    paymentCred <-
+      Helpers.liftM
+        ( error $
+            "Unable to get payment credential from Address"
+        ) $
+        addressPaymentCred address
+    pure $ stakeCredentialToKeyHash paymentCred <#> wrap
```

Comment:
Seems strange that we're dealing with stake credentials here. Seems to stem from `Internal.Serialization.Address` using `StakeCredential` to [represent both payment and delegation credentials](https://github.com/Plutonomicon/cardano-transaction-lib//blob/c2dde03df26b16501961851b959dba00faae06cf/src/Internal/Serialization/Address.purs#L372-L373). Maybe it's just different naming than I'm used to

Reply:
Originates from https://github.com/Emurgo/cardano-serialization-lib/commit/9e86bff63c1c87c7fc6736c2a7c620f9c2b7fae8
We reuse the names CSL uses, and CSL seems to prefer using types defined in the CDDL. I'm not sure if that's the right choice on their end but whatever